### PR TITLE
RPC & Suggest address

### PR DESCRIPTION
**Pull request by SAMDevelopment**
[MP-2180](https://myparcelcombv.atlassian.net/browse/MP-2180)

### CODE REVIEW

File: content/api/rpc-endpoints/_index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+title = "RPC endpoints"
+description = "MyParcel.com API RPC endpoints."
+weight = 6
++++
+
+The MyParcel.com API consists mostly of RESTful endpoints that work with resources as described in the [json-api](https://jsonapi.org) standard. However, there are situations where it makes more sense to step away from that standard. In these cases RPC endpoints might be a more fitting solution. These endpoints allow you to call an action on the API for a specific purpose that would not make sense in a resource oriented structure.
+
+## RPC structure
+
+RESTful endpoints consists solely of nouns. But RPC endpoints take the following form:
```
Comment (nessy) [inline content/api/rpc-endpoints/_index.md]: I think this can be one sentence.

File: content/api/rpc-endpoints/suggest-addess.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      "country_code": "NL",
+      "postal_code": "2131 BC",
+      "street_number": "679",
+      "city": "Hoofddorp",
+      "street_name": "Hoofdweg",
+      "street_number_suffix": "A1"
+    }
+  ]
+}
+```
```
Comment (nessy) [inline content/api/rpc-endpoints/suggest-addess.md]: I miss something about what will be returned is there is no address found.

File: content/api/rpc-endpoints/_index.md (hunk 1 above)
Comment (NickVries) [reply content/api/rpc-endpoints/_index.md]: Maybe reword this to:
`While RESTful endpoints consist solely of nouns, RPC endpoints take the following form:`

File: content/api/rpc-endpoints/suggest-addess.md (hunk 2 above)
Comment (SAMDevelopment) [reply content/api/rpc-endpoints/suggest-addess.md]: Good point. @denniskoster do you know what such a response would look like?

File: content/api/rpc-endpoints/suggest-addess.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+{{< icon fa-file-text-o >}}[API specification](https://docs.myparcel.com/api-specification#/RPC/post_suggest_address)
+
+To make sure the correct address is being used for a shipment, the `suggest-address` endpoint allows you to verify and possibly suggest the address of a shipment. This endpoints differs a bit from the other endpoints, since it is an [RPC endpoint](/api/rpc-endpoints).
+
+## Request
+
+The attributes that are required for an address suggestion differ per country. The endpoint will respond with a `422 Unprocessable Entity` when not all required attributes are supplied for the country and will list all required attributes.
+
+**Scope:** `addresses.suggest`
```
Comment (NickVries) [inline content/api/rpc-endpoints/suggest-addess.md]: Required Scope?

File: content/api/rpc-endpoints/suggest-addess.md (hunk 2 above)
Comment (NickVries) [reply content/api/rpc-endpoints/suggest-addess.md]: An empty array right?

Comment (denniskoster) [reply content/api/rpc-endpoints/suggest-addess.md]: Yes

File: content/api/rpc-endpoints/suggest-addess.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+POST /suggest-address HTTP/1.1
+Example: https://sandbox-api.myparcel.com/suggest-address
+```
+
+```json
+{
+  "data": {
+    "country_code": "NL",
+    "postal_code": "2131 BC",
+    "street_number": "679",
```
Comment (denniskoster) [inline content/api/rpc-endpoints/suggest-addess.md]: We have decided upon changing this to an integer

File: content/api/rpc-endpoints/suggest-addess.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+## Response
+
+The API will try to suggest all possible matches using the data provided.
+
+| Attribute              | Type   | Required |
+|------------------------|--------|----------|
+| `country_code`         | string | ✓        |
+| `postal_code`          | string |          |
+| `street_number`        | string |          |
```
Comment (denniskoster) [inline content/api/rpc-endpoints/suggest-addess.md]: We have decided upon changing this to an integer

File: content/api/rpc-endpoints/suggest-addess.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+The attributes that are required for an address suggestion differ per country. The endpoint will respond with a `422 Unprocessable Entity` when not all required attributes are supplied for the country and will list all required attributes.
+
+**Scope:** `addresses.suggest`
+
+| Attribute              | Type   | Required |
+|------------------------|--------|----------|
+| `country_code`         | string | ✓        |
+| `postal_code`          | string |          |
+| `street_number`        | string |          |
```
Comment (denniskoster) [inline content/api/rpc-endpoints/suggest-addess.md]: We have decided upon changing this to an integer
